### ISSUE TITLE
helm-map is overrided when using 2 sources and one of them have no keymap

### ISSUE THREAD

**Issue by thierryvolpiatto** [emacs-helm/helm#1680]
Here a recipe:
```elisp
(helm
 :sources `(,(helm-build-sync-source "test1"
                                     :candidates '(d e f)
                                     )
             ,(helm-build-sync-source "test2"
                                      :candidates '(a b c)
                                      :keymap (let ((map (make-sparse-keymap)))
                                                (set-keymap-parent map helm-map)
                                                (define-key map (kbd "C-c n") (lambda ()
                                                                                (interactive)
                                                                                (message "Keymap is working!")))
                                                map)))
 :buffer "*Helm test*")
```
## Expected behavior

When swiching back and forth from one source to the other `C-c n` should be available only in second source.

## Actual behavior from emacs-helm.sh if possible

Once switched to second source, when back to first source `C-c n` is available

**Comment by thierryvolpiatto** [comment]
One way to fix this is using a default value when building source which have been done in commit d8708168 but this break again keymaps in helm-mode (issue #700), so I have reverted it.

**Comment by thierryvolpiatto** [comment]
keymap in helm-comp-read was still specified as local map, fixed by 096a152b.
